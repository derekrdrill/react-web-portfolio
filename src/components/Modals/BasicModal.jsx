import React, { useContext } from 'react';
import styled from 'styled-components';
import { Button, Modal } from '@mui/material';

import { DarkLightModeContext } from '../DarkLightMode/context/DarkLightModeContext';

export const BasicModal = ({
  backdropOpacity,
  buttonVariant,
  cancelButtonText,
  children,
  handleClose,
  handleSubmit,
  hideBackdrop,
  open,
  submitButtonText,
}) => {
  const { darkMode } = useContext(DarkLightModeContext);

  return (
    <Modal
      hideBackdrop={hideBackdrop}
      onClose={handleClose}
      open={open}
      BackdropProps={{ style: { opacity: backdropOpacity ?? 0.9 } }}
    >
      <ModalContainer darkMode={darkMode}>
        <ExitButtonContainer>
          <Button color='error' onClick={handleClose} size='small' variant='contained'>
            X
          </Button>
        </ExitButtonContainer>
        <ContentContainer>{children}</ContentContainer>
        <ActionButtonsContainer>
          <ButtonContainer>
            <CancelButton fullWidth onClick={handleClose} variant={buttonVariant ?? 'contained'}>
              {cancelButtonText ?? 'Cancel'}
            </CancelButton>
          </ButtonContainer>
          <ButtonContainer>
            <Button
              color='info'
              fullWidth
              onClick={() => {
                handleSubmit();
                handleClose();
              }}
              variant={buttonVariant ?? 'contained'}
            >
              {submitButtonText ?? 'Submit'}
            </Button>
          </ButtonContainer>
        </ActionButtonsContainer>
      </ModalContainer>
    </Modal>
  );
};

// const ModalStyled = styled(Modal)(({ backdropColor }) => [
//   backdropColor && {
//     '.MuiBackdrop-root': {
//       backgroundColor: backdropColor,
//       opacity: 0.7,
//     },
//   },
// ]);

const ModalContainer = styled.div(({ darkMode }) => ({
  'h1, h2, h3, h4, h5, h6, p': {
    color: darkMode && 'black',
  },
  backgroundColor: darkMode ? 'grey' : 'white',
  marginLeft: '25%',
  marginTop: '18%',
  maxWidth: '50%',
  padding: 15,
  borderRadius: 5,
}));

const ActionButtonsContainer = styled.div({
  display: 'flex',
  justifyContent: 'space-around',
});

const ButtonContainer = styled.div({
  width: 185,
  maxWidth: 185,
});

const CancelButton = styled(Button)(({ variant }) => ({
  '&.MuiButton-root': {
    ':hover': {
      backgroundColor: variant === 'outlined' ? '#f0f0f0' : '#424242',
    },
    backgroundColor: variant === 'outlined' ? 'transparent' : '#696969',
    borderColor: variant === 'outlined' ? 'grey' : 'transparent',
    color: variant === 'outlined' ? 'grey' : 'white',
  },
}));

const ContentContainer = styled.div({
  padding: 30,
  display: 'flex',
  justifyContent: 'center',
});

const ExitButtonContainer = styled.div({
  display: 'flex',
  justifyContent: 'flex-end',
});
