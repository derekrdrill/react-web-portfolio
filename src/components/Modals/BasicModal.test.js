import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import { BasicModal } from './BasicModal';

const renderer = new ShallowRenderer();

describe('Basic Modal tests', () => {
  it('renders correctly', () => {
    renderer.render(<BasicModal open />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
