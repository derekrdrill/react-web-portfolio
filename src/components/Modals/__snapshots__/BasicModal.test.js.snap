// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic Modal tests renders correctly 1`] = `
<ForwardRef(Modal)
  BackdropProps={
    Object {
      "style": Object {
        "opacity": 0.9,
      },
    }
  }
  open={true}
>
  <styled.div
    darkMode={false}
  >
    <styled.div>
      <Styled(Component)
        color="error"
        size="small"
        variant="contained"
      >
        x
      </Styled(Component)>
    </styled.div>
    <styled.div />
    <styled.div>
      <styled.div>
        <Styled(Component)
          fullWidth={true}
          variant="contained"
        >
          Cancel
        </Styled(Component)>
      </styled.div>
      <styled.div>
        <ForwardRef(Button)
          color="info"
          fullWidth={true}
          onClick={[Function]}
          variant="contained"
        >
          Submit
        </ForwardRef(Button)>
      </styled.div>
    </styled.div>
  </styled.div>
</ForwardRef(Modal)>
`;
